api_app/analyzers_manager/observable_analyzers/ip2location.py

```diff
@@ -0,0 +1,64 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+
+class Ip2location(classes.ObservableAnalyzer):
+    base_url: str = "https://api.ip2location.io/"
+    ip2location_api_key: str
+    ip2location_api_version: str
```

Comment:
there is no need to add ip2location at the start of these attributes, plus the secret "api_key" must be preprended by `_` as stated in the documentation.

Reply:
I am having one issue there. Even though have the _api_key_name value present in the parameter still it's not working.
![WhatsApp Image 2024-01-22 at 11 56 30 AM](https://github.com/intelowlproject/IntelOwl/assets/92683836/f4636a5b-8588-4327-b73b-898d3d59bac4)
![image](https://github.com/intelowlproject/IntelOwl/assets/92683836/68ccba6f-7ec2-47c7-b3fb-74335d7cbb1f)

Reply:
what does that error shown up in the IntelOwl report means to you?

Reply:
I changed the variable name locally to _api_key_name.
Even when I did not changed it it was showing ip2location_api_key.
My apologies for not mentioning about the change :)

Reply:
somehow it's not getting the value of _api_key_name from the plugin config, Do we need to initalise it something?